orc/src/main/java/org/apache/iceberg/orc/TypeConversion.java

```diff
@@ -35,19 +41,21 @@
    * @return the ORC schema
    */
   public static TypeDescription toOrc(Schema schema,
-                                      ColumnIdMap columnIds) {
-    return toOrc(null, schema.asStruct(), columnIds);
+                                      Map<TypeDescription, IcebergColumn> columnIds) {
```

Comment:
I don't like to modify function parameters. Can you either return both from the method (use `Pair` if you need) or separate the mapping and the conversion to separate methods?